pyam/core.py

```diff
@@ -147,28 +147,58 @@ def variables(self, include_units=False):
         else:
             return pd.Series(self.data.variable.unique(), name='variable')
 
-    def append(self, other, inplace=False, **kwargs):
-        """Import or read timeseries data and append to IamDataFrame
+    def append(self, other, ignore_meta_conflict=False, inplace=False,
+               **kwargs):
+        """Append any castable object to this IamDataFrame.
+        Columns in `other.meta` that are not in `self.meta` are always merged,
+        duplicate region-variable-unit-year rows raise a ValueError.
 
         Parameters
         ----------
         other: pyam.IamDataFrame, ixmp.TimeSeries, ixmp.Scenario,
         pd.DataFrame or data file
-            an IamDataFrame, TimeSeries or Scenario (requires `ixmp`),
-            or pd.DataFrame or data file with IAMC-format data columns
+            An IamDataFrame, TimeSeries or Scenario (requires `ixmp`),
+            pandas.DataFrame or data file with IAMC-format data columns
+        ignore_meta_conflict : bool, default False
+            If False and `other` is an IamDataFrame, raise an error if
+            any meta columns present in `self` and `other` are not identical.
         inplace : bool, default False
-            if True, do operation inplace and return None
+            If True, do operation inplace and return None
         """
         ret = copy.deepcopy(self) if not inplace else self
 
         if not isinstance(other, IamDataFrame):
             other = IamDataFrame(other, **kwargs)
+            ignore_meta_conflict = True
+
+        diff = other.meta.index.difference(ret.meta.index)
+        intersect = other.meta.index.intersection(ret.meta.index)
 
-        # check that any model/scenario is not yet included in IamDataFrame
-        ret.meta = ret.meta.append(other.meta, verify_integrity=True)
+        # merge other.meta columns not in self.meta for existing scenarios
+        if not intersect.empty:
+            # if not ignored, check that overlapping meta dataframes are equal
+            if not ignore_meta_conflict:
+                cols = [i for i in other.meta.columns if i in ret.meta.columns]
+                pd.testing.assert_frame_equal(ret.meta.loc[intersect, cols],
```

Comment:
I would place this in a `try/except` so that we can provide the user with a specific error message. and/or use [`df.equals()` ](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.equals.html)

Reply:
changed to using `df.equals` and a specific error message with a list of model-scenarios that have conflicts in `meta`